core/storage/question/gae_models.py

```diff
@@ -74,17 +76,52 @@ def _get_new_id(cls):
             'The id generator for QuestionModel is producing too many '
             'collisions.')
 
+    def _trusted_commit(
+            self, committer_id, commit_type, commit_message, commit_cmds):
+        """Record the event to the commit log after the model commit.
+
+        Note that this extends the superclass method.
+
+        Args:
+            committer_id: str. The user_id of the user who committed the
+                change.
+            commit_type: str. The type of commit. Possible values are in
+                core.storage.base_models.COMMIT_TYPE_CHOICES.
+            commit_message: str. The commit description message.
+            commit_cmds: list(dict). A list of commands, describing changes
+                made in this model, which should give sufficient information to
+                reconstruct the commit. Each dict always contains:
+                    cmd: str. Unique command.
+                and then additional arguments for that command.
+        """
+        super(QuestionModel, self)._trusted_commit(
+            committer_id, commit_type, commit_message, commit_cmds)
+
+        committer_user_settings_model = (
+            user_models.UserSettingsModel.get_by_id(committer_id))
+        committer_username = (
+            committer_user_settings_model.username
+            if committer_user_settings_model else '')
+
+        question_commit_log = QuestionCommitLogEntryModel.create(
+            self.id, self.version, committer_id, committer_username,
+            commit_type, commit_message, commit_cmds,
+            constants.ACTIVITY_STATUS_PUBLIC, False
+        )
+        question_commit_log.question_id = self.id
+        question_commit_log.put()
+
     @classmethod
     def create(
-            cls, question_data, question_data_schema_version,
-            language_code):
+            cls, question_state_data, language_code, version):
         """Creates a new QuestionModel entry.
 
         Args:
-            question_data: dict. A dict representing the question data.
-            question_data_schema_version: int. The schema version for the data.
+            question_state_data: State. An object representing the question
```

Comment:
This can't be right. It's a dict right? You can't pass an oppia domain object into an ndb storage model.

Reply:
Yeah, done.